### PR TITLE
Payment ui mobile

### LINKED ISSUE #588: Implement Payment mobile screen
https://invis.io/J9NS9DP2CHE
https://invis.io/XCNS9EPTGZP
https://invis.io/Y4NS9F5Z5Q7

### LINKED ISSUE #589: Implement Pay Lightning Confirm mobile screen
https://invis.io/P2NS9FNN7VD

### LINKED ISSUE #590: Implement Pay Bitcoin mobile screen
https://invis.io/KRNS9GYT2YN
https://invis.io/PHNS9I06G2Y
https://invis.io/FPNS9IBB8DZ

### LINKED ISSUE #591: Implement Pay Bitcoin Confirm mobile screen
https://invis.io/PHNS9IOGDQB
https://invis.io/FBNS9JPKTWM

### PR DESCRIPTION
**Pull request by tanx**
Closes #588, closes #589, closes #590, closes #591

### CODE REVIEW

File: src/view/pay-lightning-confirm-mobile.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const PayLightningConfirmView = ({ store, nav, payment }) => (
+  <Background color={color.purple}>
+    <Header color={color.purple}>
+      <BackButton onPress={() => nav.goPay()} />
+      <Title title="Lightning Payment">
+        <LightningBoltIcon height={12} width={6.1} />
+      </Title>
+      <CancelButton onPress={() => nav.goHome()} />
+    </Header>
+    <MainContent>
```
Comment (valentinewallace) [inline src/view/pay-lightning-confirm-mobile.js]: Seems to be missing the copy: https://projects.invisionapp.com/share/P2NS9FNN7VD#/screens/299894025 "You are about to send a bitcoin payment over the lightning network"

File: src/view/payment-mobile.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    </Header>
+    {store.payment.useScanner ? (
+      <QRCodeScanner
+        onQRCodeScanned={({ data }) => payment.readQRCode({ data })}
+      />
+    ) : (
+      <PaymentCard store={store} payment={payment} />
+    )}
+    <View style={styles.btnWrapper}>
+      <GlasButton onPress={() => payment.pasteAddress()} style={styles.btnLeft}>
```
Comment (valentinewallace) [inline src/view/payment-mobile.js]: I can't paste the "normal" way (i.e. long press on the input field causing a paste option to pop up) on the SE. However, I can on the 8+. /shrug

File: src/view/payment-mobile.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+const PaymentCard = ({ store, payment }) => (
+  <MainContent>
+    <Card>
+      <FormText>
+        Paste the Lightning Payment Request or the Bitcoin Address to which
+        you’re sending.
+      </FormText>
+      <FormStretcher>
+        <InputField
```
Comment (valentinewallace) [inline src/view/payment-mobile.js]: Input field is too wide on the 8+: https://imgur.com/a/LvvkTLZ

Fixed by [this commit](https://github.com/lightninglabs/lightning-app/pull/821/commits/04a474e2e70b2c72030ff872fb494bd683f95b72) from my invoice PR, so maybe we can disregard for now.

File: src/view/pay-bitcoin-mobile.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+              autoFocus={true}
+              value={store.payment.amount}
+              onChangeText={amount => payment.setAmount({ amount })}
+              onSubmitEditing={() => nav.goPayBitcoinConfirm()}
+            />
+            <BalanceLabelUnit style={styles.unit}>
+              {store.unitFiatLabel}
+            </BalanceLabelUnit>
+          </BalanceLabel>
+          <InputField
```
Comment (valentinewallace) [inline src/view/pay-bitcoin-mobile.js]: My invoice PR #821 has fixes for these input fields that should close #824. It might make more sense to review that first, solidify the fixes, apply them, and only then merge this view? Then no need for #824 as well.

We could also separate this view into its own PR, since the rest of the views seem (pretty much) mergeable to me.

File: src/view/pay-bitcoin-confirm-mobile.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  totalLbl: {
+    marginTop: 5,
+  },
+  confirmBtn: {
+    marginTop: 20,
+    backgroundColor: color.orange,
+  },
+});
+
+const PayBitcoinConfirmView = ({ store, nav, payment }) => (
```
Comment (valentinewallace) [inline src/view/pay-bitcoin-confirm-mobile.js]: I really like the way this screen looks!

File: src/component/qrcode-scanner.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    backgroundColor: color.white,
+  },
+  text: {
+    color: color.blackText,
+    textAlign: 'center',
+    maxWidth: 300,
+  },
+});
+
+export default class QRCodeScanner extends React.Component {
```
Comment (valentinewallace) [inline src/component/qrcode-scanner.js]: Works great :)

File: stories/screen-story.js [hunk 7]
```diff
@@ -35,9 +35,12 @@ import Invoice from '../src/view/invoice';
 import InvoiceQR from '../src/view/invoice-qr';
 import Payment from '../src/view/payment';
```
Comment (valentinewallace) [inline stories/screen-story.js]: Missing PaymentMobile view in storybook?

File: src/view/pay-lightning-confirm-mobile.js (hunk 1 above)
Comment (tanx) [reply src/view/pay-lightning-confirm-mobile.js]: Yeah, we removed this later when we added the invoice description since the screen felt a bit crowded. It was also kind of redundant to the Title "Lightning Payment".

File: src/view/payment-mobile.js (hunk 2 above)
Comment (tanx) [reply src/view/payment-mobile.js]: Really? It worked for me.

File: stories/screen-story.js (hunk 7 above)
Comment (tanx) [reply stories/screen-story.js]: Can't render this one in storybook unfortunately since the react-native-web doesn't have Expo's BarCodeScanner component.

File: src/view/pay-bitcoin-mobile.js (hunk 4 above)
Comment (tanx) [reply src/view/pay-bitcoin-mobile.js]: Ok. I'll take a look at that one next.

**Comment by tanx** [comment]
> Main feedback is I think PayBitcoin should be in a separate PR until the input fields are fixed

The input field issue isn't limited to PayBitcoin. They pretty much happen on all screens. Since the fix will likely happen on the `InputField` component itself, those changes shouldn't be in conflict with this. PR.

> The rest of the code & screens seem more or less good to go, minor comments notwithstanding! :)

Ok cool. I think I'll go ahead and merge this then. We can fix the InputField separately later.
